src/libstd/fs.rs

```diff
@@ -343,7 +343,9 @@ pub fn read_to_string<P: AsRef<Path>>(path: P) -> io::Result<String> {
 #[stable(feature = "fs_read_write_bytes", since = "1.26.0")]
 pub fn write<P: AsRef<Path>, C: AsRef<[u8]>>(path: P, contents: C) -> io::Result<()> {
     fn inner(path: &Path, contents: &[u8]) -> io::Result<()> {
-        File::create(path)?.write_all(contents)
+        let mut file = File::create(path)?;
+        file.set_len(contents.len() as u64)?;
```

Comment:
We should probably ignore the error here, as AFAICT it's not guaranteed that this will succeed. In particular, my manpage for ftruncate64 says:

> For non-XSI-compliant systems, the POSIX standard allows two behaviors for ftruncate() when length exceeds the file  length  (note that  truncate() is not specified at all in such an environment): either returning an error, or extending the file.  Like most UNIX implementations, Linux follows  the  XSI  requirement  when dealing  with native filesystems.  However, some nonnative filesystems do not permit truncate() and ftruncate() to be used to extend a file beyond its current length:  a  notable  example  on Linux is VFAT.  


It's also true that I think we're changing the behavior slightly -- if the write_all call fails for whatever reason (e.g. network disconnect on a network file system) this will now create a zero-filled file rather than one with not yet all the bytes written. I'm not sure that this is the right thing to do.

Reply:
You do make a fair point - filling a file with zeroes would be break anything that relies on all of the written bytes being valid in the error case.

I'm closing this PR. Thanks for the feedback!